### PR TITLE
`General`: Fix course card style and improve course overview top bar

### PR DESCRIPTION
**Pull request by coaxial1**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

We've seen two issues:

1. Overflow of "due date" indicator on course cards on small screens
grafik

2. Bad responsiveness in course overview top bar
grafik
Additionally, the refresh button was not visible anymore if the Exam tab is enabled.

### Description

1. Easy fix by applying no-wrap styles.

Bildschirmfoto 2021-12-13 um 17 06 37

2. For some reason all the controls of the inner components (like the exercise list) are currently rendered by itself, and then moved up into the top bar using absolute pixel offsets. This PR renders them in the top bar natively.

https://user-images.githubusercontent.com/23171488/145895599-6f0a9f75-0a5f-4d87-bb5d-a4b30fa88845.mov

### Steps for Testing

- 1 Course with at least one Exercise that has a due date in the near future

1. Go to "Course Overview" and check that the Due Date indicator in the bottom right of the course card is rendered correctly on iPhone screen sizes (does not break into 4 lines or sth)
2. Go into the course. Verify that the Filter and Sorting buttons in the top right of the list are available and functional.
3. Scale your view port horizontally (make it smaller) and verify that responsiveness works (like in the video above)
4. Switch to "Lectures". Verify that the buttons in the bar changes to the "Sort lectures" dropdown and that it works correctly.
5. Switch to "Statistics". Verify that the "Hide optional" checkbox is rendered next to the Refresh button and that it still works.
6. Change to "Discussions". Verify that no other controls than the Refresh button are rendered in the top bar.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by willergeorg — COMMENTED** [review]
Tested on TS1. Functionality worked fine. I left you some small suggestions regarding the client tests. Also check the unused imports in course-statistics.component.ts

**Comment by coaxial1** [comment]
@willergeorg I've addressed your suggestions in dd8a269. I can't find any unused imports, though 🤔

File: src/main/webapp/app/overview/course-exercises/course-exercises.component.html [hunk 1]
```diff
@@ -1,73 +1,69 @@
...
-            <div class="course-overview-controls" *ngIf="course!.exercises && course!.exercises.length > 0">
-                <div class="d-flex align-items-center me-2">
-                    <div class="me-2" aria-label="Filter Dropdown" ngbDropdown>
-                        <button
-                            class="btn"
-                            [ngClass]="{ 'btn-secondary': activeFilters.size === 0, 'btn-success': activeFilters.size > 0 }"
-                            ngbDropdownToggle
-                            id="filter-dropdown-button"
+<ng-template #controls>
+    <div class="course-overview-controls" *ngIf="course && course!.exercises && course!.exercises.length > 0">
```
Comment (akesfeden) [inline src/main/webapp/app/overview/course-exercises/course-exercises.component.html:2]:
```suggestion
    <div class="course-overview-controls" *ngIf="course?.exercises?.length > 0">
```
Nitpick: For conciseness, optional chaining can be utilized but not necessarily required.

**Review by akesfeden — APPROVED** [review]
Tested on TS3, responsiveness is working as expected with varying screen sizes. Code also looks good, left a minor suggestion although it is not strictly necessary.

**Review by ole-ve — APPROVED** [review]
Tested on TS1, works described. 👍

File: src/main/webapp/app/overview/course-exercises/course-exercises.component.html (hunk 1 above)
Comment (coaxial1) [reply src/main/webapp/app/overview/course-exercises/course-exercises.component.html:2]: I did actually try this. But TypeScript complained about it. I don't know why :D
grafik

Comment (akesfeden) [reply src/main/webapp/app/overview/course-exercises/course-exercises.component.html:2]: Oh I see, right. [Here](https://stackoverflow.com/questions/54884488/how-can-i-solve-the-error-ts2532-object-is-possibly-undefined#:~:text=The%20following%20will%20fail%20in%20strict%20TypeScript%2C%20as%20you%20are%20possibly%20comparing%20an%20undefined%20value%20with%20a%20number.) I found out it is warning against the possible risk of having `undefined > 0`. 
This seems to work but a little implicit: `*ngIf="course?.exercises?.length"`
All in all, it isn't a critical point and I already approved anyway😄

**Review by DanielKainz — APPROVED** [review]
Tested it on TS1. Everything worked as expected :-).

**Pull request merged**